apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -126,23 +126,16 @@ message_stats() ->
 %% the broker.
 -spec session_open(emqx_types:clientid()) -> {_New :: boolean(), session_id(), [iterator_id()]}.
 session_open(ClientID) ->
-    {atomic, Ret} =
-        mria:transaction(
-            ?DS_SHARD,
-            fun() ->
-                case mnesia:read(?SESSION_TAB, ClientID) of
-                    [#session{iterators = Iterators}] ->
-                        IteratorIDs = maps:values(Iterators),
-                        {false, ClientID, IteratorIDs};
-                    [] ->
-                        Iterators = #{},
-                        Session = #session{id = ClientID, iterators = Iterators},
-                        mnesia:write(?SESSION_TAB, Session, write),
-                        {true, ClientID, _IteratorIDs = []}
-                end
-            end
-        ),
-    Ret.
+    case mnesia:dirty_read(?SESSION_TAB, ClientID) of
```

Comment:
This transaction is not unnecessary. This design can introduce a situation when the entire session gets dropped, so I would prefer to avoid premature optimizations.

Reply:
For reference: https://github.com/emqx/emqx/pull/11362#discussion_r1289240085

Currently, we are also relying on [another locking mechanism](https://github.com/emqx/emqx/blob/c84ee66d801bbe79e9e68e8cad222e778812f84d/apps/emqx/src/emqx_cm_locker.erl#L47), which is `quorum` by default.

Nevertheless, maybe we can reintroduce the transaction to ensure correctness and later look at this point if it proves to be a bottleneck, indeed.

Reply:
:heavy_check_mark: